### PR TITLE
Add unit tests

### LINKED ISSUE #3: Create a unit test and a travis job
Not only will this help to maintain the example, but it will also be nice an example of how to setup a travis job with gogoprotobuf and grpc.

### PR DESCRIPTION
**Pull request by johanbrandhorst**
Fixes #3.

### CODE REVIEW

**Comment by johanbrandhorst** [comment]
@awalterschulze doesn't look like I have permissions to add Travis CI to this repo. Could you do it?

**Comment by awalterschulze** [comment]
I sent an invitation to you, to become a member of github.com/gogo
Can you check whether this way you are able to create a travis job first?

File: .travis.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+language: go
+
+script: go test -v ./...
+
+go:
+  - "1.8.x"
+  - "1.9.x"
+  - "1.10.x"
```
Comment (awalterschulze) [inline .travis.yaml:8]: You will probably also need this
https://github.com/gogo/googleapis/blob/master/install-protobuf.sh
https://github.com/gogo/googleapis/blob/master/.travis.yml#L1-L2
To also exercise the code generation again

File: server/server_test.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+
+	pbExample "github.com/gogo/grpc-example/proto"
+	"github.com/gogo/grpc-example/server"
+)
+
+//go:generate mockgen -destination mocks_test.go -package server_test github.com/gogo/grpc-example/proto UserService_ListUsersServer,UserService_ListUsersByRoleServer
```
Comment (awalterschulze) [inline server/server_test.go:14]: First time seeing golang/mock in action.  The reflection (in the generated code) and the timing (setting up the expected before doing the call), is a bit weird, but I see that it saves a bit of boiler plate code and if this is the standard way of mocking in golang, then its good to showcase this as well.
Thanks for teaching me something :)

**Review by awalterschulze — APPROVED** [review]

File: server/server_test.go (hunk 2 above)
Comment (johanbrandhorst) [reply server/server_test.go:14]: At work we actually use [ginkgo](https://onsi.github.io/ginkgo/) and [gomega](https://onsi.github.io/gomega/) coupled with either gomock or [moq](github.com/matryer/moq). For these tests I wanted to stay reasonably close to the stdlib and I saw https://github.com/grpc/grpc-go/blob/master/Documentation/gomock-example.md which seems to favor this structure.

File: .travis.yaml (hunk 1 above)
Comment (johanbrandhorst) [reply .travis.yaml:8]: I will do this in a separate PR as currently the generation is tied to the third_party folder contents and I  want to be able to see that there were no generation changes.

Comment (johanbrandhorst) [reply .travis.yaml:8]: https://github.com/gogo/grpc-example/issues/5.

**Pull request merged**